static/app/views/dashboards/datasetConfig/errorsAndTransactions.tsx

```diff
@@ -596,9 +598,35 @@ function getEventsSeriesRequest(
     }
   }
 
+  if (shouldUseOnDemandMetrics(organization, widget)) {
+    return doOnDemandMetricsRequest(api, requestData);
+  }
+
   return doEventsRequest<true>(api, requestData);
 }
 
+async function doOnDemandMetricsRequest(
+  api,
+  requestData
+): Promise<
+  [EventsStats | MultiSeriesEventsStats, string | undefined, ResponseMeta | undefined]
+> {
+  const isEditing = location.pathname.endsWith('/edit/');
+
+  const fetchEstimatedStats = () =>
+    `/organizations/${requestData.organization.slug}/metrics-estimation-stats/`;
+
+  const response = await doEventsRequest<false>(api, {
+    ...requestData,
+    dataset: 'metricsEnhanced',
+    generatePathname: isEditing ? fetchEstimatedStats : undefined,
+  });
+
+  response[0] = {...response[0], isMetricsData: true, isExtrapolatedData: isEditing};
```

Comment:
Does `...response[0]` cause a hard crash if the request fails? Should we add some error handling?

Reply:
yep, i added a fallback to the original request